task/sbom-json-check/0.1/sbom-json-check.yaml

```diff
@@ -0,0 +1,65 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: sbom-json-check
+spec:
+  description: >-
+    Check the syntax of the sbom-cyclonedx.json file which should be found
+    in the /root/buildinfo/content_manifests/ directory
+  params:
+    - name: IMAGE_URL
+      description: the fully qualified image name to be verified
+      type: string
+  results:
+    - description: Test output
+      name: HACBS_TEST_OUTPUT
+  workspaces:
+    - name: workspace
+  steps:
+  - name: sbom-json-check
+    image: quay.io/redhat-appstudio/hacbs-test:latest
+    workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+    securityContext:
+      runAsUser: 0
+      capabilities:
+        add:
+          - SETFCAP
+    volumeMounts:
+      - mountPath: /shared
+        name: shared
+    script: |
+      cd /shared
+      touch fail_result.txt
+      if [ -f "sbom-cyclonedx.json" ]
+      then
+        result=$(echo -n $(cyclonedx-linux-x64 validate --fail-on-errors --input-file sbom-cyclonedx.json))
+        if [[ ! $result =~ "BOM validated successfully" ]]
+        then
+          echo "sbom-cyclonedx.json: $result" > fail_result.txt
+        fi
+      else
+        echo "cannot access 'sbom-cyclonedx.json': No such file or directory" > fail_result.txt
+      fi
+
+      FAIL_RESULTS="$(cat fail_result.txt)"
+      if [[ -z $FAIL_RESULTS ]]
+      then
+        HACBS_TEST_OUTPUT=$(jq -rc --arg date $(date +%s) --null-input \
+          '{result: "SUCCESS",timestamp: $date, namespace: "default", successes: 1, failures: 0}')
+      else
+        HACBS_ERROR_OUTPUT=$(jq -rc --arg date $(date +%s) --arg FAIL_RESULTS "${FAIL_RESULTS: 0: 3000}" --null-input \
+          '{result: "FAILURE", timestamp: $date, failures: [$FAIL_RESULTS]}')
+      fi
+
+      echo "${HACBS_TEST_OUTPUT:-${HACBS_ERROR_OUTPUT}}" | tee $(results.HACBS_TEST_OUTPUT.path)
+  sidecars:
+  - image: $(params.IMAGE_URL)
+    name: copy-the-tested-file
+    volumeMounts:
+    - mountPath: /shared
+      name: shared
+    script: cp -R /root/buildinfo/content_manifests/* /shared/
```

Comment:
It's probably a fringe scenario it could be that the built image doesn't have a `/usr/bin/cp` command in it, e.g. the file based catalog images or images built from scratch. Should we be concerned with this?

Reply:
I suppose that the pipeline will not contain this task and many others.